### PR TITLE
Add TSV, CSV types

### PR DESCRIPTION
**Pull request by sanmai-NL**
Also some line ordering and whitespace cleanups.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -689,30 +689,30 @@ macro_rules! mime_constant_test {
...
-    IMAGE_GIF, "image/gif", 5;
-    IMAGE_PNG, "image/png", 5;
-    IMAGE_BMP, "image/bmp", 5;
-
-    APPLICATION_JSON, "application/json", 11;
-    APPLICATION_WWW_FORM_URLENCODED, "application/x-www-form-urlencoded", 11;
-    APPLICATION_OCTET_STREAM, "application/octet-stream", 11;
-    APPLICATION_MSGPACK, "application/msgpack", 11;
-
-    MULTIPART_FORM_DATA, "multipart/form-data", 9;
```
Comment (seanmonstar) [inline src/lib.rs]: The ordering here was actually just matching the order the types are in described in the spec... Can we leave it that way for now?

**Comment by sanmai-NL** [comment]
Could you make those changes please? Not sure which spec and what the order would be with my additions.

**Comment by seanmonstar** [comment]
Merged manually in https://github.com/hyperium/mime/commit/2eea649b27e8c01cb29ebb60f689463908682e52.

Thanks!

**Comment by sanmai-NL** [comment]
@seanmonstar: I think you left out the test code, did you?

**Comment by seanmonstar** [comment]
I did. It wasn't necessary to have tests matching on the contents of every constant (though, the macro does automatically add tests that the indices in the macro are correct). Those tests are just making sure case-insensitivity is used, and the existing tests already do that.

**Pull request closed without merging**